### DISCUSSION TITLE
xrdp on NetBSD

### DISCUSSION THREAD

**Discussion by erwo** [neutrinolabs/xrdp#2733] [Q&A]
Hello, i use xrdp on FreeBSD for al long time, there where never a Problem.

ATM i try to get xrdp running on NetBSD, NetBSD has the PCVT console Driver, but in the Logs is the following Problem:

Fatal server error:
(EE) xf86OpenConsole: No console driver found
        Supported drivers: pccons (with X support), pcvt, wscons
        Check your kernel's console driver configuration and /dev entries(EE) 

Maybe anyone has an tip for me what the problem is?

**Comment by metalefty** [comment]
I'm a FreeBSD developer but not familiar with NetBSD. Unfortunately, I have no idea.  If you would like to use xrdp on NetBSD, let's get involved in development! We're open.

**Comment by matt335672** [comment]
@erwo - I suggest you try with the VNC backend first. The interface to xrdp is simpler, and it will give you a good idea what other problems you might encounter.

**Comment by erwo** [comment]
Error Messages with vnc:
[ERROR] sesman_data_in: scp_process_msg failed
[ERROR] sesman_main_loop: trans_check_wait_objs failed, removing trans
[ERROR] VNC error 1 after security negotiation

And a any-vnc connection failed to a local running vnc server:
[INFO ] loaded module 'libvnc.so' ok, interface size 4064, version 4
[INFO ] VNC started connecting
[INFO ] VNC connecting to 127.0.0.1 5901
[INFO ] VNC tcp connected 
[INFO ] VNC security level is 2 (1 = none, 2 = standard)  
[INFO ] VNC password ok
[INFO ] VNC sending share flag
[INFO ] VNC receiving server init
[INFO ] VNC receiving pixel format
[INFO ] VNC receiving name length
[INFO ] VNC error - problem connecting
[INFO ] some problem 
[ERROR] xrdp_wm_log_msg: Error connecting to user session


But run with a standard X Server would be the best.
Maybe it is only a small problem? I dont know which console X wants, just a tty in dev maybe?

**Comment by matt335672** [comment]
Those errors are consistent with the session not starting, and the X server getting killed early on.

I suspect that whatever is getting called from `/etc/xrdp/startwm.sh` (or wherever it is on NetBSD) is terminating too quickly. Try running the script by hand to see what happens.

**Comment by matt335672** [comment]
We can go back to the standard X server when you get something working with VNC. If we can't get VNC working, there's little point in worrying about Xorg.
